crates/sc-proof-of-time/src/gossip.rs

```diff
@@ -91,11 +126,17 @@ impl<Block: BlockT> Validator<Block> for PotGossipValidator {
     fn validate(
         &self,
         _context: &mut dyn ValidatorContext<Block>,
-        _sender: &PeerId,
+        sender: &PeerId,
         mut data: &[u8],
     ) -> ValidationResult<Block::Hash> {
         match PotProof::decode(&mut data) {
-            Ok(_) => ValidationResult::ProcessAndKeep(topic::<Block>()),
+            Ok(proof) => {
+                if self.pot_state.is_candidate(*sender, &proof).is_ok() {
+                    ValidationResult::ProcessAndKeep(topic::<Block>())
+                } else {
+                    ValidationResult::Discard
```

Comment:
Does this mean we do not forward it to peers?

Reply:
yes. This is just a repeat of the checks we do once the gossiped proof is received later. To break it down:
1. If bad seed/key, we anyway don't want to gossip it
2. If from the past (proof.slot_number behind current tip): we already gossiped it if its a good proof and it needs to be gossiped(the node is a clock master). Or bad proof from past, we don't want to gossip